packages/block-library/src/template-part/edit/use-template-part-post.js

```diff
@@ -26,20 +26,25 @@ export default function useTemplatePartPost( postId, slug, theme ) {
 					'postType',
 					'wp_template_part',
 					{
-						status: 'auto-draft',
+						status: [ 'publish', 'auto-draft' ],
```

Comment:
This now allows the TP placeholder block to find matches for CPT TPs. :tada:

I have a very small PR open (#21623) where I removed this status field from the query as well as a css rule noted below in order to get CPT TPs previewable/insertable through the placeholder.  It may make sense for me to close that out in favor of what you are doing here, but in the process of looking into that I found a few other things that may be worth noting while we are in this area (but not necessarily anything blocking this PR):

1.  Inserting 'broke' recently.  In the PR I mentioned inserting currently works as expected.  As soon as I rebase it with master (or try it on this branch since it is more recent), inserting now inserts a blank version of the TP instead (which may load properly if the editor is reloaded after saving its parent entity).  I looked around a little to see what change caused this but did not really have any luck.

2.  As far as the placeholder/inserter is concerned, I can't seem to get this to find any matches for theme supplied template parts.  After testing a little, it seems `getEntityRecord` with the provided Id will retrieve the theme supplied TP while `getEntityRecords` with a query will not.  I could find the twentytwenty header in 'auto-draft' status on the singular selector after knowing the Id for it, but the results from the plural selector with this query seem to ignore it. :disappointed:

3. It seems the preview of the template parts from the placeholder/inserter is cut off at the bottom.  You can really notice it if you create one that is only a couple words in a paragraph block and then try to insert it with the placeholder component.  Removing https://github.com/WordPress/gutenberg/blob/c6fb7f9247abcf95e282b9f8ef652a47558a838a/packages/block-library/src/template-part/editor.scss#L19-L21 seems to fix the issue for me.  Maybe we could consider removing that here (if that change makes sense to you) if Im closing out that other PR.

Reply:
>Inserting 'broke' recently. In the PR I mentioned inserting currently works as expected. As soon as I rebase it with master (or try it on this branch since it is more recent), inserting now inserts a blank version of the TP instead (which may load properly if the editor is reloaded after saving its parent entity). I looked around a little to see what change caused this but did not really have any luck.

This appears to be an inner blocks regression. #21804

>As far as the placeholder/inserter is concerned, I can't seem to get this to find any matches for theme supplied template parts. After testing a little, it seems getEntityRecord with the provided Id will retrieve the theme supplied TP while getEntityRecords with a query will not. I could find the twentytwenty header in 'auto-draft' status on the singular selector after knowing the Id for it, but the results from the plural selector with this query seem to ignore it. 😞

I just tried `header` and `twentytwenty-blocks` in this PR and it worked as expected.

>It seems the preview of the template parts from the placeholder/inserter is cut off at the bottom. You can really notice it if you create one that is only a couple words in a paragraph block and then try to insert it with the placeholder component...

I think it makes more sense to update your PR to be just that; it's unrelated to this PR.

Reply:
Thanks for the clarification!

> I just tried header and twentytwenty-blocks in this PR and it worked as expected.

Oh it is great to see this working!  I may have been testing it wrong, I saw a record for 'header' and 'twentytwenty' when I was looking before that I got though the Id, so I had been trying to get that one to show up.  But testing on this PR with the block based themes works!  🎉

Reply:
Nice!

I just found the cause of the insertion regression. I'll ping you on the write-up :)